openstf/model/objective.py

```diff
@@ -78,49 +92,135 @@ def __call__(self, trial: optuna.trial.FrozenTrial) -> float:
         # Configure evals for early stopping
         eval_set = [(train_x, train_y), (valid_x, valid_y)]
 
-        model = ModelCreator.create_model(self.model_type)
-
-        params = self.get_params(trial)
+        # get the parameters used in this trial
+        hyper_params = self.get_params(trial)
 
-        model.set_params(**params)
+        # insert parameters into model
+        self.model.set_params(**hyper_params)
 
-        eval_metric = self.eval_metric
+        # create the specific pruning callback
+        pruning_callback = self.get_pruning_callback(trial)
 
-        # See https://optuna.readthedocs.io/en/stable/reference/generated/optuna.integration.XGBoostPruningCallback.html
-        pruning_callback = optuna.integration.XGBoostPruningCallback(
-            # Use validation_0 to prune on the train set
-            # Use validation_1 to prune on the validation set
-            trial,
-            observation_key=f"validation_1-{eval_metric}",
-        )
         # validation_0 and validation_1 are available
-        model.fit(
+        self.model.fit(
             train_x,
             train_y,
             eval_set=eval_set,
             early_stopping_rounds=EARLY_STOPPING_ROUNDS,
             verbose=self.verbose,
-            eval_metric=eval_metric,
+            eval_metric=self.eval_metric,
             callbacks=[pruning_callback],
         )
 
-        forecast_y = model.predict(test_x)
+        forecast_y = self.model.predict(test_x)
+
+        return self.eval_metric_function(test_y, forecast_y)
+
+    def get_default_params(self, trial: optuna.trial.FrozenTrial) -> dict:
+        # Default parameters
+        params = {
+            "learning_rate": trial.suggest_float("learning_rate", 0.01, 0.2),
+            "alpha": trial.suggest_float("alpha", 1e-8, 1.0),
+            "lambda": trial.suggest_float("lambda", 1e-8, 1.0),
+            "subsample": trial.suggest_float("subsample", 0.5, 0.99),
+            "min_child_weight": trial.suggest_int("min_child_weight", 1, 6),
+            "max_depth": trial.suggest_int("max_depth", 3, 10),
+            "colsample_bytree": trial.suggest_float("colsample_bytree", 0.5, 1.0),
+            "max_delta_step": trial.suggest_int("max_delta_step", 1, 10),
+        }
+        return params
 
-        return mae(test_y, forecast_y)
+    def get_loss(self):
+        pass
 
 
 class XGBRegressorObjective(RegressorObjective):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.model_type = MLModelType.XGB
 
+    # extend the parameters with the model specific ones per implementation
     def get_params(self, trial: optuna.trial.FrozenTrial) -> dict:
+        default_params = super().get_default_params(trial)
+
+        # Compare the list to the default parameter space
+        model_parameters = self.model.get_params()
```

Comment:
I'm not sure if I understand this part? Why are the parameters we want to optimize per model type not static? Or are some of the default parameters not used in all models?